src/loading.jl

```diff
@@ -43,8 +43,8 @@ function parse_pkg_files(id::PkgId)
             @assert reqs !== nothing
             pkgdata.requirements = reqs
             for chi in includes
-                if isdefined(Base, :loaded_precompiles) && haskey(Base.loaded_precompiles, id => buildid)
-                    mod = Base.loaded_precompiles[id => buildid]
+                if isdefined(Base, :maybe_loaded_precompile) && Base.maybe_loaded_precompile(id, buildid) isa Module
+                    mod = Base.maybe_loaded_precompile(id, buildid)
```

Comment:
```suggestion
                if isdefined(Base, :maybe_loaded_precompile) && Base.maybe_loaded_precompile(id, buildid) isa Module
                    mod = Base.maybe_loaded_precompile(id, buildid)
                elseif isdefined(Base, :loaded_precompiles) && haskey(Base.loaded_precompiles, id => buildid)
                    mod = Base.loaded_precompiles[id => buildid]
```

To avoid breaking 1.11